server/src/test/java/org/apache/calcite/test/ServerParserTest.java

```diff
@@ -116,6 +115,30 @@ class ServerParserTest extends SqlParserTest {
         .ok("CREATE TABLE `X` (`I` INTEGER NOT NULL, `J` VARCHAR(5))");
   }
 
+  /**
+   * Test case for <a href="https://issues.apache.org/jira/browse/CALCITE-6275">[CALCITE-6275]
+   * Parser for data types ignores element nullability in collections</a>. */
+  @Test void testCreateTableWithArray() {
+    sql("create table x ("
+        + "a int array, "
+        + "b int array not null, "
+        + "c int not null array, "
+        + "d int not null array not null, "
+        + "e int array null, " // same as column a
+        + "f int null array null, " // same as column a
+        + "g int null array not null, " // same as column b
+        + "h int not null array null)") // same as column c
+        .ok("CREATE TABLE `X` ("
+            + "`A` INTEGER ARRAY, "
+            + "`B` INTEGER ARRAY NOT NULL, "
+            + "`C` INTEGER NOT NULL ARRAY, "
```

Comment:
Is  `INTEGER NOT NULL ARRAY`supported in Postgresql?

Reply:
you are right, it does not seem to be supported.
I will discuss this topic in the JIRA.